pkg/apis/forklift/v1beta1/plan.go

```diff
@@ -44,6 +44,11 @@ type PlanSpec struct {
 	TransferNetwork *core.ObjectReference `json:"transferNetwork,omitempty"`
 	// Whether this plan should be archived.
 	Archived bool `json:"archived,omitempty"`
+	// LUKs keys for virt-v2v (https://www.libguestfs.org/virt-v2v.1.html)
+	// References a secret containing credentials and
+	// other confidential information.
+	// +optional
+	LUKS core.ObjectReference `json:"luks" ref:"Secret"`
```

Comment:
based on the discussion on #873, we better be consistent here and place this per-VM rather than at the plan level
1. to be aligned with other per-VM configuration - hooks, primary disk
2. it's more flexible, allowing users to specify different keys for different VMs that are migrated to the same namespace without requiring different plans (while still specifying the keys at the UI-level per-plan to ease scale migrations)

Reply:
As I wrote in the related PR, I think LUKs make more sense as a plan, since we have the `all` option and to have the "mass migration" idea of MTV.
Your points are also valid.

Reply:
> since we have the `all` option

the `all` argument just means that the encryption keys would apply to all devices
by setting the encryption keys per VMs, we would still be able to say "for VM A, use encryption keys X, while for VM B, use encryption keys Y" (in other words, we get a better granularity)

> and to have the "mass migration" idea of MTV

it wouldn't conflict with the mindset of mass migration if the UI would provide similar functionality to that of hooks (no one is going to write yaml for mass migration, it would likely be generated by a script and then it doesn't really matter whether to place it in one place or per-VM)

@liranr23 in my opinion, we should think about it differently - the functional benefits of placing it per-VM are clear, do you see any functional benefit of placing it at the plan level?

Reply:
no, i was in mind it's theoretically fits better with the overall idea of the migrations.  
let's go with per-vm, seems it's more beneficial and aligns with other properties.